File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.py
import sys
input = sys.stdin.readline

N = int(input())
cnt = 0
for _ in range(N):
    word = input().rstrip()
    stack = []
    # 연속된 문자가 아니고, 이전에 나온 문자이면 ㄴㄴ
    for i in word:
        if not stack or i not in stack:
            stack.append(i)
        # 연속된 문자
        elif stack and i in stack and stack[-1] == i:
            continue
        # 연속된 문자가 아니고, 이전에 나온 문자
        else:
            break
    else:
        cnt += 1

print(cnt)

File: 백준/Silver/1316. 그룹 단어 체커/그룹 단어 체커.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

const n = Number(input[0]);
let result = n;

for (let i = 1; i < n + 1; i++) {
  const word = input[i];

  for (let j = 0; j < word.length; j++) {
    if (word.slice(j + 1).includes(word[j]) && word[j + 1] !== word[j]) {
      result -= 1;
      break;
    }
  }
}

console.log(result);
